tsuru/admin/broker.go

```diff
@@ -0,0 +1,211 @@
+package admin
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/tsuru/gnuflag"
+	"github.com/tsuru/go-tsuruclient/pkg/client"
+	"github.com/tsuru/go-tsuruclient/pkg/tsuru"
+	"github.com/tsuru/tablecli"
+	"github.com/tsuru/tsuru/cmd"
+)
+
+type BrokerAdd struct {
+	broker tsuru.ServiceBroker
+	fs     *gnuflag.FlagSet
+}
+
+func (c *BrokerAdd) Info() *cmd.Info {
+	return &cmd.Info{
+		Name:    "service-broker-add",
+		Usage:   "service-broker-add <name> <url> [-i/--insecure] [-c/--context key=value] [-t/--token token] [-u/--user username] [-p/--password password]",
+		Desc:    `Adds a new Service Broker.`,
+		MinArgs: 2,
+	}
+}
+
+func (c *BrokerAdd) Flags() *gnuflag.FlagSet {
+	if c.fs == nil {
+		c.fs = flagsForServiceBroker(&c.broker)
+	}
+	return c.fs
+}
+
+func (c *BrokerAdd) Run(ctx *cmd.Context, cli *cmd.Client) error {
+	apiClient, err := client.ClientFromEnvironment(&tsuru.Configuration{
+		HTTPClient: cli.HTTPClient,
+	})
+	if err != nil {
+		return err
+	}
+	c.broker.Name, c.broker.URL = ctx.Args[0], ctx.Args[1]
+	_, err = apiClient.ServiceApi.ServiceBrokerCreate(context.TODO(), c.broker)
+	if err != nil {
+		return err
+	}
+	fmt.Fprintln(ctx.Stdout, "Service broker successfully added.")
+	return nil
+}
+
+type BrokerUpdate struct {
+	broker tsuru.ServiceBroker
+	fs     *gnuflag.FlagSet
+}
+
+func (c *BrokerUpdate) Info() *cmd.Info {
+	return &cmd.Info{
+		Name:    "service-broker-update",
+		Usage:   "service-broker-updates <name> <url> [-i/--insecure] [-c/--context key=value] [-t/--token token] [-u/--user username] [-p/--password password]",
+		Desc:    `Updates a service broker.`,
+		MinArgs: 2,
+	}
+}
+
+func (c *BrokerUpdate) Flags() *gnuflag.FlagSet {
+	if c.fs == nil {
+		c.fs = flagsForServiceBroker(&c.broker)
+	}
+	return c.fs
+}
+
+func (c *BrokerUpdate) Run(ctx *cmd.Context, cli *cmd.Client) error {
+	apiClient, err := client.ClientFromEnvironment(&tsuru.Configuration{
+		HTTPClient: cli.HTTPClient,
+	})
+	if err != nil {
+		return err
+	}
+	c.broker.Name, c.broker.URL = ctx.Args[0], ctx.Args[1]
+	_, err = apiClient.ServiceApi.ServiceBrokerUpdate(context.TODO(), c.broker.Name, c.broker)
+	if err != nil {
+		return err
+	}
+	fmt.Fprintln(ctx.Stdout, "Service broker successfully updated.")
+	return nil
+}
+
+type BrokerDelete struct {
+	fs *gnuflag.FlagSet
+}
+
+func (c *BrokerDelete) Info() *cmd.Info {
+	return &cmd.Info{
+		Name:    "service-broker-delete",
+		Usage:   "service-broker-delete <name>",
+		Desc:    `Removes a service broker.`,
+		MinArgs: 1,
+	}
+}
+
+func (c *BrokerDelete) Flags() *gnuflag.FlagSet {
```

Comment:
The Flags method is not necessary if there are no flags. (It's in a separated optional interface.)